mods/taskbar-dock-animation.wh.cpp

```diff
@@ -0,0 +1,436 @@
+
+void ResetAllIconScales(std::vector<TaskbarIconInfo>& icons) {
+    if (icons.empty()) return;
+    try {
+        for (auto& iconInfo : icons) {
+            auto transform = iconInfo.element.RenderTransform().try_as<ScaleTransform>();
+            if (transform) {
+                transform.ScaleX(1.0);
+                transform.ScaleY(1.0);
+            }
+        }
+    } catch (winrt::hresult_error const& e) {
+        Wh_Log(L"DockAnimation: HRESULT error in ResetAllIconScales: %s", e.message().c_str());
+    }
+}
+
+// Event handlers
+void OnTaskbarPointerMoved(void* pThis_key, Input::PointerRoutedEventArgs const& args) {
+    try {
+        auto it = g_contexts.find(pThis_key);
+        if (it == g_contexts.end()) return;
+        auto& ctx = it->second;
+
+        if (!ctx.isInitialized) {
+            RefreshIconPositions(ctx);
+            ctx.isInitialized = true;
+        }
+        if (ctx.icons.empty()) return;
+
+        double mouseX = args.GetCurrentPoint(ctx.taskbarFrame).Position().X;
+        ApplyAnimation(mouseX, ctx);
+    } catch (winrt::hresult_error const& e) {
+        Wh_Log(L"DockAnimation: HRESULT Error in OnTaskbarPointerMoved: %s", e.message().c_str());
+    }
+}
+
+void OnTaskbarPointerExited(void* pThis_key) {
+    try {
+        auto it = g_contexts.find(pThis_key);
+        if (it == g_contexts.end()) return;
+
+        ResetAllIconScales(it->second.icons);
+        it->second.isInitialized = false;
+        it->second.icons.clear();
+    } catch (winrt::hresult_error const& e) {
+        Wh_Log(L"DockAnimation: HRESULT Error in OnTaskbarPointerExited: %s", e.message().c_str());
+    }
+}
+
+// Discover taskbar buttons and cache their centers
+void FindAllTaskbarButtonsRecursive(FrameworkElement element, DockAnimationContext& ctx) {
+    auto className = winrt::get_class_name(element);
+
+    if (std::wstring_view(className).find(L"TaskListButton") != std::wstring_view::npos) {
+        try {
+            if (!element.RenderTransform() || !element.RenderTransform().try_as<ScaleTransform>()) {
+                auto scaleTransform = ScaleTransform();
+                scaleTransform.CenterX(0.5);
+                scaleTransform.CenterY(0.5);
+                element.RenderTransform(scaleTransform);
+            }
+            element.RenderTransformOrigin({ 0.5, 1.0 });
+
+            auto transform = element.TransformToVisual(ctx.taskbarFrame);
+            auto point = transform.TransformPoint({ 0, 0 });
+
+            TaskbarIconInfo info;
+            info.element = element;
+            info.originalCenterX = point.X + (element.ActualWidth() / 2.0);
+            ctx.icons.push_back(info);
+        } catch (...) {}
+    }
+
+    int childrenCount = Media::VisualTreeHelper::GetChildrenCount(element);
+    for (int i = 0; i < childrenCount; i++) {
+        auto child = Media::VisualTreeHelper::GetChild(element, i).try_as<FrameworkElement>();
+        if (child) FindAllTaskbarButtonsRecursive(child, ctx);

```

Comment:
Instead of going over all elements recursively, wouldn't it be faster and easier to just go over all the child elements of `TaskbarFrameRepeater`?